### ISSUE TITLE
Repeat public ID queries causes overload and request rejections

### ISSUE THREAD

**Issue by dtheobald1960** [Metaswitch/crest#256]
During my testing I'll do stuff like this:

* create subscriber(s)
* query public IDs with chunk-proportions of 1, 256 and 500
* delete subscriber(s)
* repeat with a different number of subscribers

After a couple of iterations of this type of activity, I start seeing stuff like this in homestead-prov's log:

```
06-10-2015 18:58:34.401 UTC INFO base.py:187: Accepted 90.000000 requests, latency error = 7.433691, HSS overloads = 0, decrease rate 12.948170 to 10.790142
06-10-2015 18:58:34.417 UTC INFO base.py:256: Received request from localhost - PUT http://http_homestead_prov/private/9723946703@dat.richlab.datcon.co.uk
06-10-2015 18:58:34.420 UTC INFO base.py:269: Sending 200 response to localhost for PUT http://http_homestead_prov/private/9723946703@dat.richlab.datcon.co.uk
06-10-2015 18:58:34.431 UTC INFO base.py:256: Received request from localhost - POST http://http_homestead_prov/irs
06-10-2015 18:58:34.432 UTC INFO base.py:269: Sending 201 response to localhost for POST http://http_homestead_prov/irs
06-10-2015 18:58:34.443 UTC INFO base.py:256: Received request from localhost - PUT http://http_homestead_prov/private/9723946703@dat.richlab.datcon.co.uk/associated_implicit_registration_sets/c27a199f-bf9e-4b25-8196-e1fec849e1de
06-10-2015 18:58:34.443 UTC WARNING base.py:258: Rejecting request because of overload
06-10-2015 18:58:34.444 UTC INFO base.py:269: Sending 503 response to localhost for PUT http://http_homestead_prov/private/9723946703@dat.richlab.datcon.co.uk/associated_implicit_registration_sets/c27a199f-bf9e-4b25-8196-e1fec849e1de
06-10-2015 18:58:34.494 UTC INFO base.py:256: Received request from localhost - GET http://http_homestead_prov/private/9723946703@dat.richlab.datcon.co.uk
06-10-2015 18:58:34.496 UTC INFO base.py:269: Sending 200 response to localhost for GET http://http_homestead_prov/private/9723946703@dat.richlab.datcon.co.uk
06-10-2015 18:58:34.535 UTC INFO base.py:256: Received request from localhost - GET http://http_homestead_prov/public/sip:9723946703@dat.richlab.datcon.co.uk/service_profile
06-10-2015 18:58:34.535 UTC WARNING base.py:258: Rejecting request because of overload
06-10-2015 18:58:34.536 UTC INFO base.py:269: Sending 503 response to localhost for GET http://http_homestead_prov/public/sip:9723946703@dat.richlab.datcon.co.uk/service_profile
```

The homestead-prov under test is idle except for this test, so I know it's not overloaded, yet I see the above errors. Perhaps there's a miscalculation? I've eyeballed top during this and the node appears idle as one would expect with such a low amount of activity.

Diags are available.

**Comment by rkd-msw** [comment]
I suspect this is because our load monitor has a target latency per request of 100ms, but retrieving all subscribers will usually take longer.

We should probably bypass the load monitor for requests to query all public IDs. Maybe move the function in https://github.com/Metaswitch/crest/blob/dev/src/metaswitch/crest/api/base.py#L262 out into a ThrottlingHandler, and have everything except the all-IDs URL inherit from that?

**Comment by rkd-msw** [comment]
https://github.com/Metaswitch/crest/pull/259 should have fixed this - back to you for retest.

**Comment by sathiyan-sivathas** [comment]
Looks like this has been fixed and tested under #259 so I'll close this.
